### PR TITLE
[CALCITE-6321] Add copy(List<RexLiteral>) method to Window class

### PR DESCRIPTION
**Pull request by normanj-bitquill**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/interpreter/Bindables.java [hunk 1]
```diff
@@ -836,6 +836,11 @@ public static class BindableWindow extends Window implements BindableRel {
           constants, getRowType(), groups);
     }
 
+    @Override public Window copy(List<RexLiteral> constants) {
```
Comment (jduo) [inline core/src/main/java/org/apache/calcite/interpreter/Bindables.java:839]: There should be a comment that indicates that this _replaces_ (not adds) the existing constants on the node with the supplied constants.

**Review by jduo — COMMENTED** [review]
Let's add some simple unit tests for this.

File: core/src/main/java/org/apache/calcite/interpreter/Bindables.java (hunk 1 above)
Comment (normanj-bitquill) [reply core/src/main/java/org/apache/calcite/interpreter/Bindables.java:839]: Added a Javadoc comment in the `Window` class.

**Comment by normanj-bitquill** [comment]
@jduo Added some unit tests. They are cumbersome to write. Are they worth keeping?

**Comment by normanj-bitquill** [comment]
I'll move this PR to Apache.

**Comment by normanj-bitquill** [comment]
@jduo  Here is the Apache PR.

https://github.com/apache/calcite/pull/3736

**Comment by jduo** [comment]
> @jduo Added some unit tests. They are cumbersome to write. Are they worth keeping?

Let's spend some time looking to see if there are existing test classes that have more of the infrastructure built up already. Usually there are classes with a fixture that does most of the setup.

**Pull request closed without merging**